scripts/scr_marine_struct/scr_marine_struct.gml

```diff
@@ -0,0 +1,371 @@
+			var edits = variable_struct_get_names(selec_trait);
+			
+			//loop over stats and add stats where needed
+			var stats = ["constitution", "strength", "luck", "dexterity", "wisdom", "piety", "charisma", "technology"]
+			for (var stat_iter =0; stat_iter <array_length(stats);stat_iter++;){
+				if (array_contains(edits ,stats[stat_iter])){
+					 variable_struct_set(self,stats[stat_iter],  (variable_struct_get(self, stats[stat_iter])+  variable_struct_get(selec_trait, stats[stat_iter])));
+				}
+			}
+			//max_health = 100 * (1+((constitution - 20)*0.05));
+			array_push(traits, trait);
+	}
+	
+	//takes dict and plumbs dict values into unit struct
+	if (array_contains(variable_struct_get_names(global.base_stats), class)){
+		load_json_data(global.base_stats[$ class]);
+	}
+	
+
+	switch class{
+		case "marine":				//basic marine class //adds specific mechanics not releveant to most units
+			if (faction ="chapter"){
+				allegiance = global.chapter_name;
+			}
+			if (instance_exists(obj_controller)){
+				role_history = [[obj_ini.role[company,marine_number], obj_controller.turn]]; //marines_promotion and demotion history
+				marine_ascension = obj_controller.turn; // on what day did turn did this marine begin to exist
+			} else {
+				role_history = [[obj_ini.role[company,marine_number], "pre_game"]];
+				marine_ascension = "pre_game"; // on what day did turn did this marine begin to exist
+
+			}
+			if (irandom(99)>98){
+				 add_trait("very_hard_to_kill");	 //chance for marine to be exceedingly tough
+			};
+			if (irandom(999)>998){
+				 add_trait("paragon");				//paragon chance just like cm
+			};
+			if (irandom(49)>48){
+				 add_trait("warp_touched");			//has phychic potential
+			};		
+			if (allegiance == "Lamentors"){		//lamentors are unlucky
+				if (irandom(3)>2){
+					add_trait("shitty_luck");
+				}
+			}
+			break;
+		case "skitarii":
+			break;
+			
+		
+	}
+		static race = function(){return obj_ini.race[company,marine_number];}		//get race
+		static experience =  function(){return obj_ini.experience[company,marine_number];}//get exp
+		static update_exp = function(new_val){obj_ini.experience[company,marine_number] =new_val}//change exp
+		static bionics = function(){return obj_ini.bio[company,marine_number];}// get marine bionics count
+		static update_bionics = function(new_val){obj_ini.bio[company,marine_number] = new_val;}
+		static age = function(){return obj_ini.age[company,marine_number];}// age
+		static update_age = function(new_val){obj_ini.age[company,marine_number] = new_val;}		
+		static name = function(){return obj_ini.name[company,marine_number];}// get marine name
+				
+		max_health = 100 * (1+((constitution - 20)*0.05));
+		static increase_max_health = function(increase){
+			max_health += (increase*(1+((constitution - 20)*0.05))); //calculate the effect of health buffs
+		}
+		static hp = function(){ 
+			return obj_ini.hp[company,marine_number] //return current health
+		};
+		//update_marine health
```

Comment:
reduntant comment, 1 line below we see `update_health`

Reply:
done